### ISSUE TITLE
Filter pages from social shareds

### ISSUE THREAD

**Issue by greggh** [BinaryMoon/wp-toolbelt#17]
I love that the social share buttons just show up everywhere by default. I want it on 90% of my site. But there are a few pages that I need it to be hidden on. I could do this with CSS specific to those pages. Wouldn't it be better to have a list of page ids to filter out?

**Comment by greggh** [comment]
It might not be possible to do it just by ID. I am using BuddyPress on this site, and the buttons appear on all of the edit, invite, and other pages that really shouldn't be shared. These pages are dynamic, so the buttons don't even have a url. If I go to the Email Invites tab for a BuddyPress user, the share buttons appear at the bottom looking all beautiful, but their url is like this:

`https://facebook.com/sharer/sharer.php?u=`

**Comment by BinaryMoon** [comment]
Thanks - this makes total sense. I shall change it so it's blog posts by default, and so that you can filter the pages it appears on if you want it to appear elsewhere.

**Comment by greggh** [comment]
Looking things over, I like that it tries to appear everywhere. CSS can deal with that for me. But the big problem is that it doesn't handle dynamic URLs. If that could be fixed, then everything else is awesome.

**Comment by BinaryMoon** [comment]
Well... I've now changed the plugin so it's all filterable. So now it will only display on single posts, and toolbelt projects by default.

If you want it to display everywhere then you can use a filter to do that. I've added an example in the docs.
https://github.com/BinaryMoon/wp-toolbelt/wiki/Static-Social-Sharing#display-social-sharing-on-all-single-post-types

**Comment by BinaryMoon** [comment]
The issue with the url being empty must be to do with the fact that the url is dynamic.

I think I should just quit and not display anything in these situations. does that sound right? If there's no direct permalink then it's not possible to share them.

**Comment by greggh** [comment]
Not at all, more than 90% of the shareable pages on my site are dynamic. They need share buttons.

Gregg Housh
CEO
MacGuffin Ventures
m: 978-495-0776
w: macguffinventures.com e: gregg@macguffinventures.com

**Comment by BinaryMoon** [comment]
Well that's interesting. Do you have a link to a problem page so I can see what's happening for myself? Perhaps then I can work out how to recreate the problem locally and then I can fix it more easily.

**Comment by greggh** [comment]
All buddypress group pages, all by press forum topics, a lot of the WooCommerce pages, and most of the membership plugins.

The version of the plugin I have installed from earlier shows up on some of them, not all of them. But on all of them it has the missing share url.

Yo recreate it, do a default install, then buddypress, then create a group and look at the group. The one I have shows up at the bottom of the page and looks great, just the missing url issue.

**Comment by BinaryMoon** [comment]
I tested Woocommerce and it seemed to work fine, but I saw the problem with buddypress.

I shall generate the url manually if it doesn't exist (using SERVER properties) - that should cover all bases.

Keep in mind that you will now have to use this small snippet to display the share buttons on every page. I know that's some extra code but I felt hiding it by default better reflected the most common use cases.
https://github.com/BinaryMoon/wp-toolbelt/wiki/Static-Social-Sharing#display-social-sharing-on-all-single-post-types
